pgmpy/tests/test_factors/test_Factor.py

```diff
@@ -99,6 +118,17 @@ def test_marginalize(self):
         self.phi1.marginalize(['x3'])
         np_test.assert_array_equal(self.phi1.values, np.array(66))
 
+    def test_marginalize_hashable(self):
```

Comment:
@khalibartan @abhi95 Don't use `1` or `2` in function names. Come up with something that describes what the test is for.